projects/crud/src/lib/components/model-form/model-form.component.ts

```diff
@@ -161,13 +162,17 @@ export class ModelFormComponent implements OnInit, OnDestroy {
     }
 
     onAction(link) {
-        switch (link.action) {
-            case 'iframe':
-                this.openIframe(link);
-                break;
-            case 'request':
-                this.requestAction(link);
-                break;
+        if (link.action == 'iframe') {
+            this.openIframe(link);
+        }
+        else if (link.action == 'request' && link.type == 'scan') {
+            this.fillFormControls(link);
+        }
+        else if (link.action == 'dialog') {
+            this.openOpentecPopup(link);
```

Comment:
@lamiselzobeidy please make this more generalized, if the type is dialog we can call a function called open action dialog, and generalize the dialog itself